executor/mem_reader.go

```diff
@@ -265,30 +272,29 @@ func (m *memTableReader) getMemRows(ctx context.Context) ([][]types.Datum, error
 	return m.addedRows, nil
 }
 
-func (m *memTableReader) decodeRecordKeyValue(key, value []byte) ([]types.Datum, error) {
+func (m *memTableReader) decodeRecordKeyValue(key, value []byte, resultRows *[]types.Datum) error {
 	handle, err := tablecodec.DecodeRowKey(key)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return errors.Trace(err)
 	}
-	return m.decodeRowData(handle, value)
+	return m.decodeRowData(handle, value, resultRows)
 }
 
 // decodeRowData uses to decode row data value.
-func (m *memTableReader) decodeRowData(handle kv.Handle, value []byte) ([]types.Datum, error) {
+func (m *memTableReader) decodeRowData(handle kv.Handle, value []byte, resultRows *[]types.Datum) error {
```

Comment:
This is the C-style code, since Go support multiple return values, passing *pointer to parameter as return value is not necessary.
Instead, you can write code like this:

```
func (m *memTableReader) decodeRowData(handle kv.Handle, value []byte, buf []types.Datum) ([]types.Datum, error) {
        ...
        buf = append(buf, xxx)
        ...
        return buf
}

buf := make([]types.Datum, 0, len(size))
decodeRowData(..., buf)
```

Reply:
Done, PTAL @tiancaiamao